translation/message_format.rst

```diff
@@ -0,0 +1,436 @@
+        ? 0
+        : ((($number % 10 >= 2)
+            && ($number % 10 <= 4)
+            && (($number % 100 < 10)
+            || ($number % 100 >= 20)))
+                ? 1
+                : 2
+    );
+
+In order to facilitate this, the possible cases in the ``plural`` function are
+also different for each language. For instance, Russian has ``one``, ``few``,
+``many`` and ``other``, while English has only ``one`` and ``other``. The full
+list of possible cases can be found in Unicode's `Language Plural Rules`_
+document. By prefixing with ``=``, you can match exact values (like ``0`` in
+the above example).
+
+Usage of this string is the same as with variables and select::
+
+    // ...
+
+    // prints "There is one apple..."
+    echo $translator->trans('nr_of_apples', ['apples' => 1]);
+
+    // prints "There are 23 apples!"
+    echo $translator->trans('nr_of_apples', ['apples' => 23]);
+
+.. note::
+
+    You can also set an ``offset`` variable to determine whether the
+    pluralization should be offset (e.g. in sentences like ``You and # other people``
+    / ``You and # other person``). 
+
+.. tip::
+
+    When combining the ``select`` and ``plural`` functions, try to still have
+    ``select`` as outermost function:
+
+    .. code-block:: text
+
+		{gender_of_host, select, 
+            female {
+                {num_guests, plural, offset:1 
+                =0    {{host} does not give a party.}
+                =1    {{host} invites {guest} to her party.}
+                =2    {{host} invites {guest} and one other person to her party.}
+                other {{host} invites {guest} and # other people to her party.}}
+            }
+            male {
+                {num_guests, plural, offset:1 
+                =0    {{host} does not give a party.}
+                =1    {{host} invites {guest} to his party.}
+                =2    {{host} invites {guest} and one other person to his party.}
+                other {{host} invites {guest} and # other people to his party.}}
+            }
+            other {
+                {num_guests, plural, offset:1 
+                =0    {{host} does not give a party.}
+                =1    {{host} invites {guest} to their party.}
+                =2    {{host} invites {guest} and one other person to their party.}
+                other {{host} invites {guest} and # other people to their party.}}
+            }
+        }
+
+Other Placeholder Functions
+---------------------------
+
+Besides these, the MessageFormat comes with a couple other interesting functions.
+
+Ordinal
+~~~~~~~
+
+Similair to ``plural``, ``selectordinal`` allows you to use numbers as ordinal scale:
+
+.. configuration-block::
+
+    .. code-block:: yaml
+
+        # translations/messages.en.yaml
+        finish_place: >
+            You finished {place, selectordinal,
+                one {#st}
+                two {#nd}
+                few {#rd}
+                other {#th}
+            }!
```

Comment:
Can you explain why to use the `!` here at the end? Or did I miss something? 🤔

Reply:
Part of the message ("You finished first!"). If you think it's confusing I'll remove it :)

Reply:
Oh right, na let’s keep it 👍🏻